sys/include/clif.h

```diff
@@ -0,0 +1,218 @@
+ */
+ssize_t clif_encode_link(clif_t *link, char *buf, size_t maxlen);
+
+/**
+ * @brief Decodes a string of link format
+ *
+ * @param[out] links        array of links to populate
+ * @param[in]  links_len    length of @p links
+ * @param[out] params       array of params to populate
+ * @param[in]  params_len   length of @p params
+ * @param[in]  buf          string to decode
+ * @param[in]  maxlen       size of @p buf
+ *
+ * @return amount of decoded links in success
+ * @return CLIF_NOT_FOUND if the string is malformed
+ */
+unsigned clif_decode_links(clif_t *links, unsigned links_len,
+                           clif_param_t *params, unsigned params_len,
+                           char *buf, size_t maxlen);
+
+/**
+ * @brief   Adds a given @p target to a given buffer @p buf using link format
+ *
+ * @param[in]  target  string containing the path to the resource
+ * @param[out] buf     buffer to output the formatted path. Can be NULL
+ * @param[in]  maxlen  size of @p buf
+ *
+ * @note If @p buf is NULL this will return the amount of bytes that would be
+ *       needed
+ *
+ * @return in success the amount of bytes used in the buffer
+ * @return CLIF_NO_SPACE if there is not enough space in the buffer
+ */
+ssize_t clif_add_target(const char *target, char *buf, size_t maxlen);
+
+/**
+ * @brief   Adds a given @p parameter to a given buffer @p buf using link format
+ *
+ * @param[in] param     pointer to the parameter to add
+ * @param[in] buf       buffer to add the parameter to. Can be NULL
+ * @param[in] maxlen    size of @p buf
+ *
+ * @note If @p buf is NULL this will return the amount of bytes that would be
+ *       needed
+ *
+ * @return amount of bytes used from the buffer if successful
+ * @return CLIF_NO_SPACE if there is not enough space in the buffer
+ */
+ssize_t clif_add_param(clif_param_t *param, char *buf,
+                              size_t maxlen);
+
+/**
+ * @brief   Adds the link separator character to a given @p buf, using link
+ *          format
+ *
+ * @param[in] buf       buffer to add the separator to. Can be NULL
+ * @param[in] maxlen    size of @p buf
+ *
+ * @note If @p buf is NULL this will return the amount of bytes that would be
+ *       needed
+ *
+ * @return amount of bytes used from buffer if successful
+ * @return CLIF_NO_SPACE if there is not enough space in the buffer
+ */
+ssize_t clif_add_link_separator(char *buf, size_t maxlen);
+
+/**
+ * @brief   Looks for a link in a given string @p input. If multiple link are
+ *          present in the string it will return the first one.
+ *
+ * @param[in] input         string where to look for a link
+ * @param[in] input_len     length of @p input
+ * @param[out] output       if a link is found this will point to the beginning
+ *                          of it
+ *
+ * @return length of the link if found
+ * @return CLIF_NOT_FOUND if no valid link is found
+ */
+ssize_t clif_get_link(const char *input, size_t input_len,
```

Comment:
Is this function useful?  Why not either use `clif_decode_link()` or `clif_get_target()`?

Reply:
It is being used in `clif_decode_links` to find the link separator and know the size of each one. When `clif_get_param` if called the string should be limited to one link, that's why this function exists.